src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/StartupProjectRegistrar.cs

```diff
@@ -111,21 +113,29 @@ protected override Tasks.Task DisposeCoreAsync(bool initialized)
 
         private async Tasks.Task AddOrRemoveProjectFromStartupProjectList(bool initialize = false)
         {
-            bool isDebuggable = await IsDebuggable().ConfigureAwait(false);
-
-            if (initialize || isDebuggable != _isDebuggable)
+            await _lock.WaitAsync().ConfigureAwait(false);
+            try
```

Comment:
This makes my brain hurt. Just assign and read the value on the UI thread, and you never have to block or lock.